src/compile/mark/valueref.ts

```diff
@@ -201,6 +202,20 @@ export function midPoint(
   return isFunction(defaultRef) ? defaultRef() : defaultRef;
 }
 
+export function tooltipForChannelDefs(channelDefs: FieldDef<string>[], config: Config) {
+  const keyValues: string[] = [];
+  const usedKey = {};
+  channelDefs.forEach(fieldDef => {
```

Comment:
I prefer `for (const fieldDef of channelDef) {...}` because it's a real for loop that you can use break and return with.